pkg/deploy/deploy.go

```diff
@@ -62,20 +62,30 @@ func DeployConfigs(client client.Client, apis api.APIs, sortedConfigs []config.C
 		var entity parameter.ResolvedEntity
 		var deploymentErrors []error
 
-		switch t := c.Type.(type) {
-
-		case config.EntityType:
-			log.Debug("Entity are not deployable, skipping entity type: %s", t.EntitiesType)
-			continue
-
-		case config.SettingsType:
-			entity, deploymentErrors = deploySetting(client, entityMap, &c)
+		properties, errs := resolveProperties(&c, entityMap.get())
+		if len(errs) > 0 {
+			entity = parameter.ResolvedEntity{}
+			deploymentErrors = append(deploymentErrors, errs...)
+		}
 
-		case config.ClassicApiType:
-			entity, deploymentErrors = deployConfig(client, apis, entityMap, &c)
+		renderedConfig, err := c.Render(properties)
+		if err != nil {
+			entity = parameter.ResolvedEntity{}
+			deploymentErrors = append(deploymentErrors, err)
+		}
 
-		default:
-			deploymentErrors = []error{fmt.Errorf("unknown config-type (ID: %q)", c.Type.ID())}
+		if deploymentErrors == nil {
+			switch t := c.Type.(type) {
+			case config.EntityType:
+				log.Debug("Entity are not deployable, skipping entity type: %s", t.EntitiesType)
+				continue
```

Comment:
\minor: Please add empty newlines between cases. I think it is way easier to distinguish cases.